### PR TITLE
Set up preprod environment/namespace for Offender Management

### PR DESCRIPTION
**Pull request by rossjones**
On the MOIC project, we have need of a preprod environment where we can
work on almost-new live production data.

### CODE REVIEW

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/elasticache.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module "ec-cluster-offender-management-allocation-manager" {
+  source                 = "github.com/ministryofjustice/cloud-platform-terraform-elasticache-cluster?ref=3.0"
```
Comment (mtrblmoj) [inline namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/elasticache.tf]: Please use the latest version of that module :

See example : https://github.com/ministryofjustice/cloud-platform-environments/pull/995/files

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/rds-allocations-api.tf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+variable "cluster_state_bucket" {}
+
+/*
+ * Make sure that you use the latest version of the module by changing the
+ * `ref=` value in the `source` attribute to the latest version listed on the
+ * releases page of this repository.
+ *
+ */
+module "allocation-rds" {
+  source = "github.com/ministryofjustice/cloud-platform-terraform-rds-instance?ref=4.0"
```
Comment (mtrblmoj) [inline namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/rds-allocations-api.tf]: please use the latest version of that module. Also, please use :
` db_instance_class      = "db.m4.large"`

example : https://github.com/ministryofjustice/cloud-platform-environments/blob/master/namespaces/live-1.cloud-platform.service.justice.gov.uk/prison-visits-booking-production/resources/rds-prison-visits.tf

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/elasticache.tf (hunk 1 above)
Comment (rossjones) [reply namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/elasticache.tf]: Done

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/rds-allocations-api.tf (hunk 2 above)
Comment (rossjones) [reply namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/rds-allocations-api.tf]: Done

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/rds-allocations-api.tf [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  db_instance_class      = "db.m4.large"
+  business-unit          = "HMPPS"
+  application            = "offender-management-allocation-manager"
+  is-production          = "true"
+  environment-name       = "preprod"
+  infrastructure-support = "omic@digital.justice.gov.uk"
+  db_engine              = "postgres"
+  db_engine_version      = "10"
+  db_name                = "allocations"
+  aws_region             = "eu-west-2"
```
Comment (mtrblmoj) [inline namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/rds-allocations-api.tf]: Replace `aws_region` with the `providers` block here please.

Example : https://github.com/ministryofjustice/cloud-platform-environments/blob/master/namespaces/live-1.cloud-platform.service.justice.gov.uk/prison-visits-booking-production/resources/rds-prison-visits.tf

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/elasticache.tf [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  source                 = "github.com/ministryofjustice/cloud-platform-terraform-elasticache-cluster?ref=3.1"
+  cluster_name           = "${var.cluster_name}"
+  cluster_state_bucket   = "${var.cluster_state_bucket}"
+  team_name              = "${var.team_name}"
+  application            = "offender-management-allocation-manager"
+  is-production          = "${var.is-production}"
+  node_type              = "cache.m4.large"
+  environment-name       = "${var.environment-name}"
+  infrastructure-support = "${var.infrastructure-support}"
+  aws_region             = "eu-west-2"
```
Comment (mtrblmoj) [inline namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/elasticache.tf]: `aws_region` is deprecated. (replaced with the `providers` block)

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/rds-allocations-api.tf (hunk 3 above)
Comment (rossjones) [reply namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/rds-allocations-api.tf]: Done

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/elasticache.tf (hunk 4 above)
Comment (rossjones) [reply namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-management-preprod/resources/elasticache.tf]: Done

**Pull request merged**